### ISSUE TITLE
Support for Gremlin Database and Gremlin Graph within Azure CosmosDB

### ISSUE THREAD

**Issue by andreasolza** [terraform-providers/terraform-provider-azurerm#4970]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

Terraform already ships with resources to access Azure CosmosDB's MongoDB, CassandraDB and Table API. However, analogous resources for GremlinDB are still missing. This should be quick to acheive.

### Proposed Resource(s)

1. azurerm_cosmosdb_gremlin_database
2. azurerm_cosmosdb_gremlin_graph

### Potential Terraform Configuration

1. Example main.tf for azurerm_cosmosdb_gremlin_database:

```
resource "azurerm_cosmosdb_gremlin_database" "main" {
  name                = var.gremlindbName
  resource_group_name = var.argMainName
  account_name        = var.cosmosdbName
}
```

2. Example main.tf for azurerm_cosmosdb_gremlin_graph:

```
resource "azurerm_cosmosdb_gremlin_graph" "main" {
  name                = var.gremlindbName
  resource_group_name = var.argMainName
  account_name        = var.cosmosdbName
  graph_name          = var.graphName

  indexing_policy {
    indexing_mode  = "consistent"
    automatic      = true
    included_paths = [{ path = "/*" }]
    excluded_paths = [{ path = "/\"_etag\"/?" }]
  }

  conflict_resolution_policy {
    mode                     = "LastWriterWins"
    conflict_resolution_path = "/_ts"
  }
}

resource "azurerm_cosmosdb_gremlin_graph_throughout" "main" {
  name                = var.gremlindbName
  resource_group_name = var.argMainName
  account_name        = var.cosmosdbName
  graph_name          = var.graphName
  throughput          = 10000
}
```

### References

https://docs.microsoft.com/en-us/azure/cosmos-db/graph-introduction
https://docs.microsoft.com/en-us/azure/cosmos-db/create-graph-gremlin-console

* #0000

**Comment by SebRosander** [comment]
I've started to look into azurerm_cosmosdb_gremlin_graph parts as well. Still a newbie in go and haven't worked with gremlin that much, so it may take some time. Bear with me.

**Comment by katbyte** [comment]
fixed by #5301
